server.js

```diff
@@ -3063,6 +3063,85 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Discourse integration
+camp.route(/^\/discourse\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  const host   = match[1]; // eg, meta.discourse.org
+  const stat   = match[2]; // eg, user_count
+  const format = match[3];
+  const url    = 'https://' + host + '/site/statistics.json';
+
+  const options = {
+    method: 'GET',
+    uri: url,
+    headers: {
+      'Accept': 'application/json'
+    }
+  };
+
+  var badgeData = getBadgeData('discourse', data);
+  request(options, function(err, res) {
+    if (err != null) {
+      if (res) {
+        console.error('' + res);
+      }
+
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    if (res.statusCode !== 200) {
+      badgeData.text[1] = 'offline';
```

Comment:
When does this case occur?

What happens for a nonexistent host? Is it possible to detect and return **host not found** for that case? A helpful error message makes it easier for users to understand what has gone wrong with their badge.

Reply:
In retrospect, I realize the **host not found** suggestion doesn't make sense. Probably this one should also be **inaccessible**.